src/Model/Contact/Relation.php

```diff
@@ -163,20 +165,22 @@ private static function updateFollowersFollowings(array $contact, array $followe
 		$following_counter = 0;
 
 		Logger::info('Discover contacts', ['id' => $target, 'url' => $url, 'contacts' => count($contacts)]);
-		foreach ($contacts as $contact) {
-			$actor = Contact::getIdForURL($contact);
-			if (!empty($actor)) {
-				if (in_array($contact, $followers)) {
-					$fields = ['cid' => $target, 'relation-cid' => $actor, 'follows' => true, 'follow-updated' => DateTimeFormat::utcNow()];
+		foreach ($contacts as $contact_url) {
+			$contact = Contact::getByURL($contact_url, false, ['id']);
+			if (!empty($contact['id'])) {
+				if (in_array($contact_url, $followers)) {
+					$fields = ['cid' => $target, 'relation-cid' => $contact['id'], 'follows' => true, 'follow-updated' => DateTimeFormat::utcNow()];
 					DBA::insert('contact-relation', $fields, Database::INSERT_UPDATE);
 					$follower_counter++;
 				}
 
-				if (in_array($contact, $followings)) {
-					$fields = ['cid' => $actor, 'relation-cid' => $target, 'follows' => true, 'follow-updated' => DateTimeFormat::utcNow()];
+				if (in_array($contact_url, $followings)) {
+					$fields = ['cid' => $contact['id'], 'relation-cid' => $target, 'follows' => true, 'follow-updated' => DateTimeFormat::utcNow()];
 					DBA::insert('contact-relation', $fields, Database::INSERT_UPDATE);
 					$following_counter++;
 				}
+			} elseif (!Network::isUrlBlocked($contact_url)) {
+				Worker::add(Worker::PRIORITY_LOW, 'AddContact', 0, $contact_url);	
 			}
```

Comment:
If this condition is supposed to be verified every time an `AddContact` worker task is created, please add the condition in an `AddContact::add()` static method the same way we do for `UpdateContact` or `UpdateGServer`.